fix: dynamic origin handling and updated redirect URLs for environmen…

OlenaReukova opened the pull request:
# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review on my code
- [ ] I have made corresponding changes to the documentation (if any were needed)
- [x] My changes generate no new warnings or errors
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and previously existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Description

**Relates #385 <u>OR</u> Closes #385 **  

Instead of defaulting to http://localhost:3000/, the **reset password email link** should dynamically generate the correct URL based on the environment.

In the deployed version, **the reset password email link** should point to the actual domain of the website (e.g., https://your-deployed-domain.com/login/reset-password) instead of the localhost URL.

**Code Update:**

- Refactored origin detection to handle cases where the value is missing or undefined smoothly.
- Default to http://localhost:3000 for development mode.
- Dynamically set the origin for production using NEXT_PUBLIC_SITE_URL environment variable.

```
const headersList = await headers();
  let origin: string | null = headersList.get('origin') || null;

  if (!origin) {
    origin =
      process.env.NODE_ENV === 'development'
        ? 'http://localhost:3000'
        : process.env.NEXT_PUBLIC_SITE_URL || null;
  }
```
**Configuration Update:**

Updated **config.toml** to include additional production redirect URLs `"https://**cool-creponne-3e1272.netlify.app/**", "https://cool-creponne-3e1272.netlify.app/login/reset-password"` to ensure Supabase accepts them, preventing failed password reset flows due to unlisted URLs.

[(Supabase Docs)](https://supabase.com/docs/guides/auth/redirect-urls)



### Files changed

- app/(auth)/login/forgot-password/page.tsx
- config.toml

### UI changes

no UI changes

### Changes to Documentation

No changes needed.

# Tests

All previous tests passed during local development.

OlenaReukova committed:
fix: dynamic origin handling and updated redirect URLs for environment compatibility

camelPhonso commented on app/(auth)/login/forgot-password/page.tsx:
```diff
@@ -10,9 +10,14 @@ export default async function ForgotPassword({
+
   if (!origin) {
-    origin = 'http://localhost:3000/';
+    origin =
+      process.env.NODE_ENV === 'development'
+        ? 'http://localhost:3000'
+        : process.env.NEXT_PUBLIC_SITE_URL || null;
```
I'm not entirely sure we need to have this `if` statement at all but definitely we don't want to evaluate to `process.env.NEXT_PUBLIC_SITE_URL` or `null` - we want to get rid of any possible `null`s here so we don't send people to a 404

camelPhonso commented on supabase/config.toml:
```diff
@@ -76,7 +76,7 @@ enabled = true
 site_url = "http://localhost:3000"
 # A list of *exact* URLs that auth providers are permitted to redirect to post authentication.
-additional_redirect_urls = ["http://localhost:3000/login/reset-password", "http://localhost:3000/**"]
+additional_redirect_urls = ["http://localhost:3000/login/reset-password", "http://localhost:3000/**", "https://**cool-creponne-3e1272.netlify.app/**", "https://cool-creponne-3e1272.netlify.app/login/reset-password"]
```
no I don't think we want this here. This field is for `additional` urls because the base url is configured on the production project in supabase to be kept out of the development environment

OlenaReukova replied:
I cancelled the changes.

camelPhonso reviewed (commented):
I have a theory that this needs to be sorted with some tweaks in supabase but I no longer have access because I'm not logged in with my old email account. Can someone reach out on discord and give me access on my personal email?

OlenaReukova committed:
fix:Cancelled changes in config.toml

OlenaReukova committed:
refactor: header origin retrieval logic

removed redundant logic for setting the origin value based on NODE_ENV and NEXT_PUBLIC_SITE_URL

Replaced with a simplified implementation: const headerData = await headers(); const origin = headerData.get('origin')

The previous fallback logic was unnecessary since the origin header is reliably retrieved using the headers() function in most environments.

OlenaReukova commented:
The console.log("Origin", origin) confirms that the origin value is fetched correctly during my tests, and no errors (e.g., null or unexpected results) occur. This makes me believe that the production deployment should work as expected, provided the server environment properly sets the origin header.

OlenaReukova committed:
Merge branch 'dev' into #385/bug-incorrect-reset-password-URL-deployed-environment

camelPhonso reviewed (approved):
Sorry for the wait Olena. Thanks for reviewing the origin set up and following up in the comments.

This seems grand 👌 ✨
